src/ConstraintSolver.jl

```diff
@@ -175,8 +181,8 @@ mutable struct ConstraintSolverModel
     backtrack_vec       :: Vector{BacktrackObj}
     sense               :: MOI.OptimizationSense
     objective           :: ObjectiveFunction
-    best_sol            :: Int # Objective of the best solution
-    best_bound          :: Int # Overall best bound
+    best_sol            :: Float64 # Objective of the best solution
```

Comment:
Should it be parameterized by `T` here, with `ConstraintSolverModel{T}`?

Reply:
In principal yes but I'm not sure how. It gets created when creating the model without knowing what gets added to it. Maybe the default is `Float64` but one can set the type with a solver option if one knows that it is always `Int`?

Reply:
When creating the model, the user could provide the type they expect

Reply:
It's a solver option now.